heartbeat/monitors/stdfields/unnest.go

```diff
@@ -72,6 +75,12 @@ func UnnestStream(config *conf.C) (res *conf.C, err error) {
 		return nil, fmt.Errorf("could not determine base stream for config: %s", id)
 	}
 
-	err = res.Merge(mapstr.M{"id": optS.Id, "data_stream": optS.DataStream})
+	res.Merge(mapstr.M{"data_stream": optS.DataStream})
+
+	// We only override the ID for the original fleet integration, not monitors configured
+	// through monitor mgmt. See https://github.com/elastic/beats/issues/32224
+	if origin == "" {
+		err = res.Merge(mapstr.M{"id": optS.Id})
+	}
```

Comment:
actually this will override in case of monitor managed monitors as well. Since in that case as well we set the origin:ui, but that's fine, since we use data streams format for monitor management and we do set the stream id as well in there. https://github.com/elastic/kibana/blob/main/x-pack/plugins/synthetics/server/synthetics_service/formatters/convert_to_data_stream.ts#L30

Reply:
That seems like a good thing, I'd like to just delete this block in the future when we fully transition